mmf/datasets/databases/readers/feature_readers.py

```diff
@@ -90,25 +92,29 @@ def read(self, image_feat_path):
 class CHWFeatureReader:
     def read(self, image_feat_path):
         if image_feat_path.endswith("npy"):
-            feat = torch.from_numpy(np.load(image_feat_path))
+            with PathManager.open(image_feat_path, "rb") as f:
+                feat = torch.from_numpy(np.load(f))
         elif image_feat_path.endswith("pth"):
-            feat = torch.load(image_feat_path, map_location=torch.device("cpu"))
+            with PathManager.open(image_feat_path, "rb") as f:
+                feat = torch.load(f, map_location=torch.device("cpu"))
         assert feat.shape[0] == 1, "batch is not 1"
         feat = feat.squeeze(0)
         return feat, None
 
 
 class Dim3FeatureReader:
     def read(self, image_feat_path):
-        tmp = np.load(image_feat_path)
+        with PathManager.open(image_feat_path, "rb") as f:
+            tmp = np.load(f)
         _, _, c_dim = tmp.shape
         image_feature = torch.from_numpy(np.reshape(tmp, (-1, c_dim)))
         return image_feature, None
 
 
 class HWCFeatureReader:
     def read(self, image_feat_path):
-        tmp = np.load(image_feat_path)
+        with PathManager.open(image_feat_path, "rb") as f:
```

Comment:
Convert this to a private utility function in this file and use it to load the field. That would be much cleaner and make it easy to make any updates to this in future.